engine/classes/Elgg/EntityIconService.php

```diff
@@ -263,22 +260,24 @@ public function saveIcon(ElggEntity $entity, ElggFile $file, $type = 'icon', arr
 			}
 
 			$icon = $this->getIcon($entity, $size, $type);
-
+			$icon->open('write');
+			$icon->close();
+			
 			// Save the image without resizing or cropping if the
 			// image size value is an empty array
 			if (is_array($opts) && empty($opts)) {
 				copy($file->getFilenameOnFilestore(), $icon->getFilenameOnFilestore());
 				continue;
 			}
 
-			$image_bytes = get_resized_image_from_existing_file($file->getFilenameOnFilestore(), $width, $height, $square, $x1, $y1, $x2, $y2, $upscale);
-			if ($image_bytes) {
-				$icon->open("write");
-				$icon->write($image_bytes);
-				$icon->close();
-				unset($image_bytes);
-			} else {
-				$this->logger->error("Failed to create {$size} icon from {$file->getFilenameOnFilestore()} with coords [{$x1}, {$y1}],[{$x2}, {$y2}]");
+			$source = $file->getFilenameOnFilestore();
+			$destination = $icon->getFilenameOnFilestore();
+
+			$resize_params = array_merge($opts, $coords);
+			
+			if (!_elgg_services()->imageService->resize($source, $destination, $resize_params)) {
```

Comment:
I am thinking we can add a batch method, so we don't have to open/close the file handle for each icon